packages/mui-system/src/createStyled.js

```diff
@@ -220,6 +220,10 @@ export default function createStyled(input = {}) {
         Component.displayName = displayName;
       }
 
+      if (tag.muiName) {
+        Component.muiName = tag.muiName;
+      }
```

Comment:
@mnajdova This is required to make `styled(Grid2)` works. Some Joy UI would benefit from this fix as well.